core/src/main/java/io/github/belgif/rest/guide/validator/core/model/OperationDefinition.java

```diff
@@ -12,7 +12,7 @@ public class OperationDefinition extends OpenApiDefinition<Operation> {
     @Setter
     private PathItem.HttpMethod method;
 
-    public OperationDefinition(Operation model, PathDefinition parent, PathItem.HttpMethod method) {
+    public OperationDefinition(Operation model, OpenApiDefinition<?> parent, PathItem.HttpMethod method) {
```

Comment:
Consider letting PathDefinition inherit from PathItemDefinition (and add them to both lists in the parser).

Reply:
I think we can further improve:
- either we make one list and two different objects (because in the rules we can easily select based on Object Type)
- we make one object, one list, with a boolean flag in the definition type: inMainFile or topLevelPath or something like that. And implement an extra filter in all the rules that collect all paths.

In my opinion the last solution might be the cleanest. What do you think?